sandbox/libs/authn/src/main/java/org/opensearch/authn/InternalSubject.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.authn;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.security.Principal;
+import java.util.Collections;
+import java.util.Map;
+
+public class InternalSubject implements Subject {
+
+    @JsonProperty(value = "primary_principal")
+    private String primaryPrincipal;
+
+    @JsonProperty(value = "hash")
+    private String bcryptHash;
+    private Map<String, String> attributes = Collections.emptyMap();
+
+    @JsonProperty(value = "primary_principal")
+    public String getPrimaryPrincipal() {
+        return primaryPrincipal;
+    }
+
+    @JsonProperty(value = "primary_principal")
+    public void setPrimaryPrincipal(String primaryPrincipal) {
+        this.primaryPrincipal = primaryPrincipal;
+    }
+
+    @JsonProperty(value = "hash")
+    public String getBcryptHash() {
+        return bcryptHash;
+    }
+
+    @JsonProperty(value = "hash")
+    public void setBcryptHash(String bcryptHash) {
+        this.bcryptHash = bcryptHash;
+    }
+
+    public Map<String, String> getAttributes() {
+        return attributes;
+    }
+
+    public void setAttributes(Map<String, String> attributes) {
+        this.attributes = attributes;
+    }
+
+    @Override
+    public String toString() {
+        return "InternalSubject [primaryPrincipal=" + primaryPrincipal + ", bcryptHash=" + bcryptHash + ", attributes=" + attributes + "]";
+    }
+
+    @Override
+    public Principal getPrincipal() {
+        return new StringPrincipal(primaryPrincipal);
+    }
+
+    @Override
+    public void login(AuthenticationToken token) {
```

Comment:
This class looks like it is (de)serializable, how does this login method work? I think the subject representation might have a reference to an object with the attributes like above.

Consider splitting this class into `InternalIdentity` and `InternalSubject`, the subject would have a reference to the identity object.  Note; there are many ways to refactor and this is only one of many possible suggestions,

Reply:
I changed this to be `User` which I think is an apt name for the class. When the basic auth handler is implemented we should use the internal realm, authenticate and connect this to the subject interface to login and create a subject